src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,246 @@
+        for the particular extractor.
+
+        Parameters
+        ----------
+        dl1_table : ndarray
+            dl1 table with images and times stored in a numpy array of shape
+            (n_images, n_channels, n_pix).
+        masked_pixels_of_sample : ndarray
+            boolean array of masked pixels that are not available for processing
+        col_name : string
+            column name in the dl1 table
+
+        Returns
+        -------
+        List StatisticsContainer:
+            List of extracted statistics and validity ranges
+        """
+        pass
+
+
+class PlainExtractor(StatisticsExtractor):
+    """
+    Extractor the statistics from a sequence of images
+    using numpy and scipy functions
+    """
+
+    def __call__(
+        self, dl1_table, masked_pixels_of_sample=None, col_name="image"
+    ) -> list:
+
+        # in python 3.12 itertools.batched can be used
+        image_chunks = (
+            dl1_table[col_name].data[i : i + self.sample_size]
+            for i in range(0, len(dl1_table[col_name].data), self.sample_size)
+        )
+        time_chunks = (
+            dl1_table["time"][i : i + self.sample_size]
+            for i in range(0, len(dl1_table["time"]), self.sample_size)
+        )
+
+        # Calculate the statistics from a sequence of images
+        stats_list = []
+        for images, times in zip(image_chunks, time_chunks):
+            stats_list.append(
+                self._plain_extraction(images, times, masked_pixels_of_sample)
+            )
+        return stats_list
+
+    def _plain_extraction(
+        self, images, times, masked_pixels_of_sample
+    ) -> StatisticsContainer:
+
+        # ensure numpy array
+        masked_images = np.ma.array(images, mask=masked_pixels_of_sample)
+
+        # median over the sample per pixel
+        pixel_median = np.ma.median(masked_images, axis=0)
+
+        # mean over the sample per pixel
+        pixel_mean = np.ma.mean(masked_images, axis=0)
+
+        # std over the sample per pixel
+        pixel_std = np.ma.std(masked_images, axis=0)
+
+        # median of the median over the camera
+        median_of_pixel_median = np.ma.median(pixel_median, axis=1)
+
+        # outliers from median
+        image_median_outliers = np.logical_or(
+            pixel_median < self.image_median_cut_outliers[0],
+            pixel_median > self.image_median_cut_outliers[1],
+        )
+
+        return StatisticsContainer(
+            validity_start=times[0],
+            validity_stop=times[-1],
+            mean=pixel_mean.filled(np.nan),
+            median=pixel_median.filled(np.nan),
+            median_outliers=image_median_outliers.filled(True),
+            std=pixel_std.filled(np.nan),
+        )
+
+
+class SigmaClippingExtractor(StatisticsExtractor):
+    """
+    Extractor the statistics from a sequence of images
+    using astropy's sigma clipping functions
+    """
+
+    sigma_clipping_max_sigma = Int(
```

Comment:
naming question: do we want/need to carry "sigma_clipping_" prefix here and on the next line? Since this is an implementation that won't have generalization, I would recommend to remove it for conciseness